src/Raven.Server/Documents/Handlers/StreamingHandler.cs

```
@@ -134,29 +134,56 @@ public async Task StreamQueryPost()
                 }
 
                 var format = GetStringQueryString("format", false);
+                var debug = GetStringQueryString("debug", false);
                 var properties = GetStringValuesQueryString("field", false);
                 var propertiesArray = properties.Count == 0 ? null : properties.ToArray();
                 
                 // set the exported file name
-                string fileName = query.Metadata.IsCollectionQuery ? query.Metadata.CollectionName + "_collection" : "query_result";
-                fileName = $"{Database.Name}_{fileName}"; 
-
-                using (var writer = GetQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
+                var fileName = query.Metadata.IsCollectionQuery ? query.Metadata.CollectionName + "_collection" : "query_result";
+                fileName = $"{Database.Name}_{fileName}";
+                if (string.IsNullOrWhiteSpace(debug) == false)
                 {
-                    try
+                    using (var writer = GetIndexEntriesQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
                     {
-                        await Database.QueryRunner.ExecuteStreamQuery(query, context, HttpContext.Response, writer, token).ConfigureAwait(false);
+                        try
+                        {
+                            await Database.QueryRunner.ExecuteStreamIndexEntriesQuery(query, context, HttpContext.Response, writer, token).ConfigureAwait(false);
+                        }
+                        catch (IndexDoesNotExistException)
+                        {
+                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                            writer.WriteError($"Index {query.Metadata.IndexName}  does not exists");
+                        }
```

We never check for the value of `debug` variable. For now we have only `"entries"`. But in the future we might have more options. Let's throw if we get unexpected value here.

---

done